crates/biome_cli/src/commands/scan_kind.rs

```diff
@@ -0,0 +1,214 @@
+use crate::Execution;
+use biome_analyze::{Queryable, RegistryVisitor, Rule, RuleDomain, RuleFilter, RuleGroup};
+use biome_configuration::Configuration;
+use biome_configuration::analyzer::{RuleDomainValue, RuleDomains};
+use biome_css_syntax::CssLanguage;
+use biome_graphql_syntax::GraphqlLanguage;
+use biome_js_syntax::JsLanguage;
+use biome_json_syntax::JsonLanguage;
+use biome_rowan::Language;
+use biome_service::workspace::ScanKind;
+use rustc_hash::FxHashSet;
+
+/// It analyzes the CLI and the configuration to understand what kind of scanning Biome needs to execute.
+///
+/// Rules:
+/// - CLI via `stdin` return [ScanKind::None]
+/// - `biome format` return [ScanKind::KnownFiles]
+/// - `biome lint`, `biome check` and `biome ci` varies. It depends on whether the user has enabled rules that require the `RuleDomain::Project`
+pub(crate) fn compute_scan_kind(execution: &Execution, configuration: &Configuration) -> ScanKind {
```

Comment:
Does this `Configuration` have CLI arguments applied to it? It would be great if users can use `--skip=...` to exclude expensive rules in combination with `--changed`, for instance, in order to selectively speed up such use cases.

Reply:
Yeah you're right, it doesn't. I probably need to use the same visitor we use in `biome_service`.

Reply:
It was relatively easy, there wasn't need of the `biome_service` visitor.